x-pack/plugins/apm/public/components/fleet_integration/apm_agents/runtime_attachment/index.tsx

```diff
@@ -75,10 +73,24 @@ export function RuntimeAttachment(props: Props) {
   const [editDiscoveryRuleId, setEditDiscoveryRuleId] = useState<null | string>(
     null
   );
-  const [version, setVersion] = useState(props.selectedVersion);
-  const [versions, setVersions] = useState(props.versions);
+  const [version, setVersion] = useState<RuntimeAttachmentSettings['version']>(
+    props.version
+  );
   const [isValidVersion, setIsValidVersion] = useState(
-    validateVersion(version)
+    validateVersion(props.version)
+  );
+
+  useEffect(
+    () => {
+      // Invalidates the package policy, so save button is disabled
+      // until a valid version is provided
+      if (isEnabled && !isValidVersion) {
+        props.invalidatePackagePolicy();
+      }
+    },
+    // props shouldn't be listed as dependency here
```

Comment:
If destructing is used to get all the props in the params can we avoid adding this eslint disable?

Reply:
not really, it creates a new reference to the function which invokes the hook.